spec/acceptance/01_class_spec.rb

```diff
@@ -3,6 +3,71 @@
 require 'spec_helper_acceptance'
 
 describe 'apptainer class:' do
+  context 'with install os package', if: fact('os.family') == 'RedHat' do
+    it 'runs successfully' do
+      pp = <<-PUPPET_PP
+      class { 'apptainer':
+        version        => 'installed',
+        install_method => 'os',
+        install_setuid => true,
+        # Avoid /etc/localtime which may not exist in minimal Docker environments
+        bind_paths     => ['/etc/hosts'],
+      }
+      PUPPET_PP
+
+      apply_manifest(pp, catch_failures: true)
+      apply_manifest(pp, catch_changes: true)
+    end
+
+    describe file('/etc/apptainer/apptainer.conf') do
+      it { is_expected.to be_file }
+      it { is_expected.to be_mode 644 }
+      it { is_expected.to be_owned_by 'root' }
+      it { is_expected.to be_grouped_into 'root' }
+    end
+
+    describe package(['apptainer', 'apptainer-suid']) do
+      it { is_expected.to be_installed }
+    end
+
+    describe command('apptainer exec docker://alpine cat /etc/alpine-release') do
+      its(:exit_status) { is_expected.to eq(0) }
+      its(:stdout) { is_expected.to match %r{[0-9]+.[0-9]+.[0-9]+} }
+    end
+  end
+
+  context 'with remove os package', if: fact('os.family') == 'RedHat' do
+    it 'runs successfully' do
+      pp = <<-PUPPET_PP
+        package{['apptainer-suid', 'apptainer']:
+          ensure => 'absent',
+        }
+        Package['apptainer-suid'] -> Package['apptainer']
+      PUPPET_PP
+
+      apply_manifest(pp, catch_failures: true)
+      apply_manifest(pp, catch_changes: true)
+    end
+
+    describe package(['apptainer', 'apptainer-suid']) do
+      it { is_expected.not_to be_installed }
+    end
+  end
+
+  context 'with singularity' do
```

Comment:
I removed the singularity tests because it's no longer possible to install singularity packages since apptainer obsoletes singularity packages.